user/user_lego_sensor.h

```diff
@@ -16,9 +16,18 @@
 #include <lego.h>
 #include <lego_sensor_class.h>
 
+/* The USER_LEGO_SENSOR_TEXT_VALUE_SIZE can be a maximum of PAGE_SIZE-2
+ * to take into account a trailing newline (added later) and NULL. We are
+ * limiting it to much less to save memory for each user_lego_sensor 
+ * instance.
+ */
+
+#define USER_LEGO_SENSOR_TEXT_VALUE_SIZE (512-2)
```

Comment:
Another thought. You could specify the buffer size in `configfs` and dynamically allocate it. This way, each userspace driver can allocate exactly what it needs and no more. If the size is `0`, then `get_text_value` returns `-EOPNOTSUPP`.